### PR TITLE
1-refactor-v1.0.1

### PR DESCRIPTION
**Pull request by jlucas9**
I did touch every single file, but ensured that all existing tests passed at each step along the way. 
Really just a lot of reformatting with the last two commits being the largest and potentially controversial as it exposes additional functionality to user space in favor of smaller files. 

Let me know what you think - if you'd rather not have one of these changes or if something was missed!  
Thanks in advance.

### CODE REVIEW

File: src/src_main/crypto_tc.c [hunk 1]
```diff
@@ -68,6 +68,10 @@ int32_t Crypto_TC_ApplySecurity(const uint8_t *p_in_frame, const uint16_t in_fra
         printf("%02X", ((uint8_t *)&*p_in_frame)[i]);
     }
     printf("\nPrinted %d bytes\n", in_frame_length);
+#else
```
Comment (szemerick) [inline src/src_main/crypto_tc.c:71]: can also do: _(void) in_frame_length;_
This method is used in FreeRTOS a lot.

**Review by szemerick — APPROVED** [review]

**Comment by IbraheemYSaleh** [comment]
There're definitely some style clashes here that I don't fully agree with (EG, enforcing the 120 maximum characters per a single line rule, forcing even simple if blocks to span multiple lines, etc), but overall, this refactor does improve the code structure for CryptoLib 👍  ... The only real issue I have is that the MariaDB code is broken with the build (-DMYSQL=ON) ... I'll go ahead and start working on fixing that and push the commits to this branch.

**Review by IbraheemYSaleh — APPROVED** [review]
With my changes (the last commit), the MariaDB code is working again and passing our downstream unit tests. I approve of this refactor now 👍 ... Please merge when ready.

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,5 @@
 .idea
 vgcore*
-core
```
[the remark this answers is not in the document]
Comment (IbraheemYSaleh) [reply .gitignore:3]: Restored in last commit.

File: src/src_main/sadb_routine_inmemory.template.c [hunk 3]
```diff
@@ -0,0 +1,975 @@
...
+ * export laws and regulations. User has the responsibility to obtain
+ * export licenses, or other export authority as may be required before
+ * exporting such information to foreign countries or providing access to
+ * foreign persons.
+ */
+
+#include "crypto.h"
+
+// Security Association Initialization Functions
+int32_t sadb_config(void);
```
[the remark this answers is not in the document]
Comment (IbraheemYSaleh) [reply src/src_main/sadb_routine_inmemory.template.c]: Resolved in last commit.

**Review by dccutrig — APPROVED** [review]

**Pull request merged**